src/webnn/native/dml/ContextDML.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "webnn/native/dml/ContextDML.h"
+
+#include "common/RefCounted.h"
+#include "webnn/native/dml/GraphDML.h"
+
+namespace webnn::native::dml {
+
+    HRESULT Context::Initialize() {
+        if (mUseDebugLayer) {
+            ComPtr<ID3D12Debug> debug;
+            if (SUCCEEDED(D3D12GetDebugInterface(IID_PPV_ARGS(&debug)))) {
+                debug->EnableDebugLayer();
+            }
+        }
+        WEBNN_RETURN_IF_FAILED(
+            D3D12CreateDevice(mAdapter.Get(), D3D_FEATURE_LEVEL_11_0, IID_PPV_ARGS(&mD3D12Device)));
```

Comment:
Will we want more than one context instance per queue (with a specified device)? If so, the queue and device should be passed by parameter, and not created within the context.

Reply:
Understood, thanks! But I think we only need one context instance per queue.

Reply:
You actually only need 1 queue for the entire process. This queue could be provided from the GPUDevice, for example.